analysis/study_definition_ethnicity.py

```diff
@@ -0,0 +1,60 @@
+from cohortextractor import (
+  StudyDefinition,
+  patients,
+  codelist,
+  codelist_from_csv,
+  Measure,
+)
+
+from datetime import date
+
+end_date = "2022-03-01"
+
+from codelists import *
+  
+study = StudyDefinition(
+    default_expectations={
+      "date": {"earliest": "1900-01-01", "latest": "today"},
+      "rate": "uniform",
+    },
+    
+    index_date=end_date,
+    
+    population = patients.satisfying(
+       """
+        NOT has_died
+        AND
+        registered
+        """,
```

Comment:
If you plan on extracting ethnicity once, and joining it to monthly extracts of data, you probably want to specify population=patients.all()
In each month the study population could be different, so if you only extract ethnicity once, you will end up with some patients missing ethnicity when we do actually have a value in the dataset. 
This would be true if you use either an analysis script join_ethnicity or the reusable action.

Reply:
Thanks for the tip, ethnicity did seem to have more missing values than expected so that explains it.